### ISSUE TITLE
It's not possible to follow invites as guest

### ISSUE THREAD

**Issue by freshprince** [vector-im/element-web#16164]
### Steps to reproduce

- Join a room as a guest
- Recieve an invite by someone else, for instance to DM
- Try to join the room you got invited to

Instead of joining the new room you get asked to register or login.

### Version information

- **Platform**: web

For the web app:

- **Browser**: Firefox
- **OS**: Linux
- **URL**: app.element.io

**Comment by t3chguy** [comment]
It is intentional that Element requires you to register to perform write actions such as joining a room (even by invite)

An Invite in Matrix speak is an explicit thing sent to your Matrix account, presumably you mean you received a matrix.to link out of band?

**Comment by freshprince** [comment]
No. I'm speaking of `/invite @some_number_since_it's_a_guest:servername`. So it is not possible to send any direct messages to guest accounts? Also the experience is a bit weird because you see the invite as the guest but are not able to do anything with it.

**Comment by jryans** [comment]
If I understand correctly, I don't think this flow is possible after https://github.com/matrix-org/matrix-react-sdk/commit/3115c4f3f3ed0b2f1d82c5882dcce007b188c206, as there shouldn't be any UI to join a room as a guest. I know that's not an ideal guest experience, but we can work that out in future ILAG etc bugs. For now, I'm just aiming to keep things consistent, so that guests are required to register when performing write actions.

Is my understanding correct, or is this flow still possible even after the room directory change?

**Comment by ericdrgn** [comment]
@jryans Sorry if this is the wrong place to ask but this seems like the same issue I am experiencing as well. I am attempting to set something up so that anyone can message me directly but you are saying this isn't possible? For instance I want someone to be able to go to a link like this https://[servername]/#/user/@eric:[server]?action=chat and them be dropped into a DM with me without having to register first. Ideally perhaps disabling registration but them still being able to do that.

Is this something that would never be possible?

**Comment by t3chguy** [comment]
Not with Element currently due to the product decision of only allowing readonly access from guests. Other clients may use them for more.

**Comment by ekpyron** [comment]
To me that seems like a rather unfortunate product decision. I'd very much prefer if Element just allowed proper guest interactions for rooms that allow guests to join.

That being said, I stumbled upon this issue because I just tried this and it's not actually properly enforced by Element right now.
While you cannot interact with rooms via invite links, a guest account can use the normal `+` sign for adding rooms and then (while rooms appearing in the search list will again lead to the read-only preview only) entering the full room ID in the text field allows properly joining via the join button that appears in the text field.
See https://app.element.io/#/room/#testcommunity42:matrix.org for reference, those messages there are I sent from guest accounts. Although those messages still suffer from https://github.com/vector-im/element-web/issues/16045.

So you might want to fix that at least, although I'd very much prefer the opposite, i.e. properly supporting interactions from guests in general :-).
